# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.cpp
# Commit: 00d83f1d02e0
# Full Hash: 00d83f1d02e015735d580045524eceeeccdc4e28
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 1 - Allow calling BrowserParent::InitRendering multiple times, and remove RenderFrame dependency from nsFrameLoader. r=kats
#   
#   This cleanup will simplify refactoring nsFrameLoader later.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31429
# ==============================================================================

diff -r bdd76368ddc1 -r 00d83f1d02e0 dom/ipc/BrowserParent.cpp
--- a/dom/ipc/BrowserParent.cpp	Wed May 22 18:37:34 2019 +0300
+++ b/dom/ipc/BrowserParent.cpp	Wed Apr 24 22:36:27 2019 -0500
@@ -884,9 +884,10 @@
 }
 
 void BrowserParent::InitRendering() {
-  MOZ_ASSERT(!mRenderFrame.IsInitialized());
+  if (mRenderFrame.IsInitialized()) {
+    return;
+  }
   mRenderFrame.Initialize(this);
-  MOZ_ASSERT(mRenderFrame.IsInitialized());
 
   layers::LayersId layersId = mRenderFrame.GetLayersId();
   AddBrowserParentToTable(layersId, this);
@@ -906,13 +907,16 @@
   frameLoader->MaybeShowFrame();
 }
 
-void BrowserParent::Show(const ScreenIntSize& size, bool aParentIsActive) {
+bool BrowserParent::Show(const ScreenIntSize& size, bool aParentIsActive) {
   mDimensions = size;
   if (mIsDestroyed) {
-    return;
+    return false;
   }
 
   MOZ_ASSERT(mRenderFrame.IsInitialized());
+  if (!mRenderFrame.AttachLayerManager()) {
+    return false;
+  }
 
   nsCOMPtr<nsISupports> container = mFrameElement->OwnerDoc()->GetContainer();
   nsCOMPtr<nsIBaseWindow> baseWindow = do_QueryInterface(container);
@@ -921,6 +925,7 @@
   mSizeMode = mainWidget ? mainWidget->SizeMode() : nsSizeMode_Normal;
 
   Unused << SendShow(size, GetShowInfo(), aParentIsActive, mSizeMode);
+  return true;
 }
 
 mozilla::ipc::IPCResult BrowserParent::RecvSetDimensions(const uint32_t& aFlags,